lib/deploy/data.ts

```diff
@@ -152,7 +153,9 @@ export async function defaultKubernetesApplication(
     context: HandlerContext,
 ): Promise<KubernetesApplication> {
 
-    const configAppData: Partial<KubernetesApplication> = _.get(k8Deploy, "sdm.configuration.sdm.k8s.app", {});
+    const k8sConfig = (k8Deploy.sdm.configuration.sdm.k8s || {}) as K8sConfiguration["k8s"];
+
+    const configAppData: Partial<KubernetesApplication> = k8sConfig.app || {};
```

Comment:
If you make app a `Partial<KubernetesApplication>`, then you do not need to have the type here.